system/Config/BaseConfig.php

```diff
@@ -82,7 +82,7 @@ public static function __set_state(array $array)
      */
     public function __construct()
     {
-        static::$moduleConfig = config(Modules::class);
+        static::$moduleConfig = new Modules();
```

Comment:
`BaseConfig` does not share the `Config\Modules`.
So there is no way to change the `Config\Modules` values during testing.

https://github.com/codeigniter4/CodeIgniter4/blob/4c09ca3cf8ab28d6df03222710a73da386dfa80b/tests/system/Config/BaseConfigTest.php#L268-L281

Reply:
The fix will be something like this, but it is a breaking change.
```diff
diff --git a/system/Config/BaseConfig.php b/system/Config/BaseConfig.php
index b4b9a4a213..2d0a0e9fe6 100644
--- a/system/Config/BaseConfig.php
+++ b/system/Config/BaseConfig.php
@@ -80,9 +80,9 @@ class BaseConfig
      *
      * The "shortPrefix" is the lowercase-only config class name.
      */
-    public function __construct()
+    public function __construct(?Modules $config = null)
     {
-        static::$moduleConfig = new Modules();
+        static::$moduleConfig = $config ?? new Modules();
 
         if (! static::$override) {
             return;
diff --git a/tests/system/Config/BaseConfigTest.php b/tests/system/Config/BaseConfigTest.php
index 2da390392e..4c66673189 100644
--- a/tests/system/Config/BaseConfigTest.php
+++ b/tests/system/Config/BaseConfigTest.php
@@ -12,6 +12,7 @@
 namespace CodeIgniter\Config;
 
 use CodeIgniter\Test\CIUnitTestCase;
+use Config\Modules;
 use Encryption;
 use RegistrarConfig;
 use RuntimeException;
@@ -267,10 +268,10 @@ final class BaseConfigTest extends CIUnitTestCase
 
     public function testNotEnabled(): void
     {
-        $modulesConfig          = config('Modules');
+        $modulesConfig          = new Modules();
         $modulesConfig->enabled = false;
 
-        $config              = new RegistrarConfig();
+        $config              = new RegistrarConfig($modulesConfig);
         $config::$registrars = [];
         $expected            = $config::$registrars;
 
@@ -282,10 +283,10 @@ final class BaseConfigTest extends CIUnitTestCase
 
     public function testDidDiscovery(): void
     {
-        $modulesConfig          = config('Modules');
+        $modulesConfig          = new Modules();
         $modulesConfig->enabled = true;
 
-        $config              = new RegistrarConfig();
+        $config              = new RegistrarConfig($modulesConfig);
         $config::$registrars = [];
         $this->setPrivateProperty($config, 'didDiscovery', false);
 
```

Reply:
We can't modify that constructor. What about a static `@testTag` setter?

Reply:
Okay. I'll change.

Reply:
Would it be okay to change `static::$moduleConfig` as nullable? phpstan complains on `??=`

Reply:
It has only `@var`, so it is okay.

https://github.com/codeigniter4/CodeIgniter4/blob/6f81e6e42e996eb1d0eacb2325600f74fefc9307/system/Config/BaseConfig.php#L56-L60